homeassistant/helpers/update_coordinator.py

```diff
@@ -99,10 +104,14 @@ def _schedule_refresh(self) -> None:
         # minimizing the time between the point and the real activation.
         # That way we obtain a constant update frequency,
         # as long as the update process takes less than a second
+        if self.last_update_success:
+            update_interval = self.update_interval
+        else:
+            update_interval = self.failed_update_interval
```

Comment:
What is stopping us from changing the update interval of the coordinator from the integration?

Reply:
The DataUpdateCoordinator schedules the next update immediately after updating the data, setting this in the integration would mean that it wouldn't work until after one more update at the regular interval

Reply:
This wouldn't accomplish the goal here, which is to provide more frequent updates when there is a failed update

Reply:
If you think this is too niche, I can extend the data update coordinator class with a custom coordinator class just for NWS.

Reply:
Other integrations do change the update interval after creation, butb for other purposes, like staying within the API call limits when a new device is added.  This PR needs to be slightly adjusted so that works correctly, but I'll wait to see if this functionality is wanted first.

Reply:
I think each integration should handle this. Different integrations need different behavior.

Reply:
Ok, I will closec this and reopen

Reply:
Can I change this to just add the last successful update time?

Reply:
We can inherit from the coordinator class and add arbitrary attributes in the integration.

If we see many integration doing the same pattern we can consider consolidating it in the future.

Reply:
Makes sense.  Thanks for considering